### PR TITLE
Xml

### PR DESCRIPTION
**Pull request by YaraslauVashkevich**
Module XML

### CODE REVIEW

File: xml/pom.xml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.11</version>
+            <scope>test</scope>
+        </dependency>
+		
+		<dependency>
```
Comment (IliaPotapovich) [inline xml/pom.xml:29]: Code formatting

File: xml/src/main/java/com/epam/upskill/Xml.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        int selected = choiceFile.showOpenDialog(null);
+        if (selected == JFileChooser.APPROVE_OPTION) {
+            File selectFile = choiceFile.getSelectedFile();
+            String path = selectFile.getAbsolutePath();
+            nameConfig = selectFile.getName();
+            logger.log(Level.INFO, "The configuration file is selected.");
+
+            XmlParsing xmlParsing = new XmlParsing();
+            //String suffix = xmlParsing.getSuffix(path);
+            //String suffix = xmlParsing.getSuffixDom(path);
```
Comment (IliaPotapovich) [inline xml/src/main/java/com/epam/upskill/Xml.java:33]: Remove redundant comments

File: xml/src/main/java/com/epam/upskill/XmlSave.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        } catch (Exception e) {
+            logger.log(Level.ERROR, "The XML file is not created", e);
+        }
+        logger.log(Level.INFO, "XML file is created.");
+    }
+
+    public void xmlSaveStax(String nameConfig, String leadTime, String oldFile, String newFile) {
+        try {
+            XMLOutputFactory outputFactory = XMLOutputFactory.newInstance();
+            //XMLStreamWriter writer = outputFactory.createXMLStreamWriter(System.out);
```
Comment (IliaPotapovich) [inline xml/src/main/java/com/epam/upskill/XmlSave.java:71]: The same

File: xml/src/main/java/com/epam/upskill/XmlSave.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            writer.writeEndElement();
+            writer.writeEndElement();
+            writer.writeEndElement();
+            writer.writeEndDocument();
+            writer.flush();
+            writer.close();
+        }catch (Exception e) {
+            logger.log(Level.ERROR, "The XML file is not created", e);
+        }
+        logger.log(Level.INFO, "XML file is created.");
```
Comment (IliaPotapovich) [inline xml/src/main/java/com/epam/upskill/XmlSave.java:102]: Add some additional information about the file (name, id or something else)

File: xml/src/main/java/com/epam/upskill/Xml.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                        newFile = newFile.append(newName + "/");
+                    }
+                }
+                logger.log(Level.INFO, "Renaming completed.");
+            }
+        }
+        String oldFiles = new String(oldFile);
+        String newFiles = new String(newFile);
+        Date data = new Date();
+        SimpleDateFormat format = new SimpleDateFormat("dd.MM.yyyy HH:mm:ss");
```
Comment (IliaPotapovich) [inline xml/src/main/java/com/epam/upskill/Xml.java:64]: A named constant field needs to be created for such cases. For example 
private static final String DATE_FORMAT = "dd.MM.yyyy HH:mm:ss";

Comment (IliaPotapovich) [reply xml/src/main/java/com/epam/upskill/Xml.java:64]: Strings in loggers stay the same (without constants)

File: src/main/resources/resultDom.xml [hunk 6]
```diff
@@ -0,0 +1 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?><result><configuration><name_config>config.xml</name_config><lead_time>03.02.2021 16:23:43</lead_time></configuration><old_file><name>1_new.txt/2_new.txt/3_new.txt/4_new.txt/5_new.txt/</name></old_file><new_file><name>1_new_new.txt/2_new_new.txt/3_new_new.txt/4_new_new.txt/5_new_new.txt/</name></new_file></result>
```
Comment (IliaPotapovich) [inline src/main/resources/resultDom.xml:1]: Why did you locate this file in the parent module?

File: src/main/resources/resultStax.xml [hunk 7]
```diff
@@ -0,0 +1 @@
+<?xml version="1.0" encoding="UTF-8"?><result><configuration><name_config>config.xml</name_config><lead_time>03.02.2021 16:23:43</lead_time></configuration><old_file><name>1_new.txt/2_new.txt/3_new.txt/4_new.txt/5_new.txt/</name></old_file><new_file><name>1_new_new.txt/2_new_new.txt/3_new_new.txt/4_new_new.txt/5_new_new.txt/</name></new_file></result>
```
Comment (IliaPotapovich) [inline src/main/resources/resultStax.xml:1]: The same

File: xml/pom.xml (hunk 1 above)
Comment (YaraslauVashkevich) [reply xml/pom.xml:29]: Reformat.

File: xml/src/main/java/com/epam/upskill/XmlSave.java (hunk 3 above)
Comment (YaraslauVashkevich) [reply xml/src/main/java/com/epam/upskill/XmlSave.java:71]: Comments deleted.

File: xml/src/main/java/com/epam/upskill/Xml.java (hunk 2 above)
Comment (YaraslauVashkevich) [reply xml/src/main/java/com/epam/upskill/Xml.java:33]: Comments deleted.

File: xml/src/main/java/com/epam/upskill/Xml.java (hunk 5 above)
Comment (YaraslauVashkevich) [reply xml/src/main/java/com/epam/upskill/Xml.java:64]: Constant field created.

File: xml/src/main/java/com/epam/upskill/XmlSave.java (hunk 4 above)
Comment (YaraslauVashkevich) [reply xml/src/main/java/com/epam/upskill/XmlSave.java:102]: Added file names.

File: src/main/resources/resultDom.xml (hunk 6 above)
Comment (YaraslauVashkevich) [reply src/main/resources/resultDom.xml:1]: The files are deleted. They were used when checking the operation of the application.

File: src/main/resources/resultStax.xml (hunk 7 above)
Comment (YaraslauVashkevich) [reply src/main/resources/resultStax.xml:1]: The files are deleted. They were used when checking the operation of the application.

**Review by IliaPotapovich — APPROVED** [review]

**Pull request merged**